sdk/sdk/OBAApplication.m

```diff
@@ -0,0 +1,120 @@
+//
+//  sdk.m
+//  sdk
+//
+//  Created by Dima Belov on 4/25/15.
+//  Copyright (c) 2015 One Bus Away. All rights reserved.
+//
+
+#import "OBAApplication.h"
+
+static NSString * const kOBAHiddenPreferenceUserId = @"OBAApplicationUserId";
+static NSString * const kOBADefaultRegionApiServerName = @"regions.onebusaway.org";
+
+@interface OBAApplication()
+
+@property(nonatomic, strong, readwrite) OBAReferencesV2 * references;
+@property(nonatomic, strong, readwrite) OBAModelDAO * modelDao;
+@property(nonatomic, strong, readwrite) OBAModelService * modelService;
+@property(nonatomic, strong, readwrite) OBALocationManager * locationManager;
+
+@end
+
+@implementation OBAApplication
+
++(instancetype) instance {
+    static OBAApplication * oba;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        oba = [[OBAApplication alloc] init];
+    });
+    
+    return oba;
+}
+
+-(void) start {
+    self.references = [[OBAReferencesV2 alloc] init];
```

Comment:
In case there is a need to async init.  I'll update the doc mentioning that only "instance" / "sharedApplication" should be used.  Same reason one would not try to re-init app delegate.